src/Microsoft.AspNetCore.Mvc.Core/DefaultMiddlewareFilterConfigurationProvider.cs

```diff
@@ -0,0 +1,151 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Globalization;
+using System.Linq;
+using System.Reflection;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.AspNetCore.Mvc
+{
+    public class DefaultMiddlewareFilterConfigurationProvider : IMiddlewareFilterConfigurationProvider
+    {
+        private readonly string _environmentName;
+
+        public DefaultMiddlewareFilterConfigurationProvider(IHostingEnvironment hostingEnvironment)
+        {
+            _environmentName = hostingEnvironment.EnvironmentName;
+        }
+
+        public void Configure(Type middlewarePipelineProviderType, IApplicationBuilder applicationBuilder)
```

Comment:
Alternatively, consider making this method have a signature like:
```
public Action<IApplicationBuilder> CreateConfigureDelegate(Type startup)
```

This is much easier to implement and limits this service to policy about how to instantiate the class and what method to call.